gradle.properties

```diff
@@ -0,0 +1 @@
+version = 2.5.1-SNAPSHOT
```

Comment:
Why 2.5.1 instead of 2.5.0? And why SNAPSHOT?

Reply:
It's conventional for Java projects to switch to a snapshot version number after releasing, and the main reason I went ahead and did it is because it allows `mavenLocal` to have less priority than `mavenCentral` (may notice that in the `repositories` block in the build.gradle.kts file). I've had many instances of confusion with the other order and having a local version of an artifact override a released one which should basically never be done.

Reply:
Ok makes sense, but still have a couple questions:

* If the main reason for adding the SNAPSHOT is to distinguish it from the version already released in `mavenCentral`, why do we also increment the patch version number? Why not just make it 2.5.0-SNAPSHOT?
* If we want to depend on the local version of the repo, say to test local-only changes with a sample app, we would depend on the SNAPSHOT version right?
* When making the next release, say 2.6.0, would we remove the SNAPSHOT suffix before releasing and then replace it afterward?

Reply:
- Yup, the snapshot is not production and just to provide previews, as long as the number is > than the previous released version and <= the next release, it works out as an interim build. So it's simple to just increment the patch number for SNAPSHOT even if the release ends up being a minor

- Yeah would depend on the SNAPSHOT version, and I believe for local no need to worry about update policy, that's for fetching snapshots from remote

- Yeah most libraries use maven release plugin or gradle release plugin which automates that. Will see if it works with our release process, and if for some reason it doesn't I'll revert to just using a normal version number :)